soc/arm/nxp_imx/rt5xx/soc.c

```diff
@@ -512,6 +513,20 @@ void __weak imxrt_deinit_display_interface(void)
 
 #endif
 
+static int nxp_rt500_late_init(void)
+{
+	/* Declare pin configuration state for isp pin here */
+	PINCTRL_DT_DEFINE(DT_NODELABEL(isp));
+
+	/* Apply pinctrl state directly, since there is no isp device driver */
+	pinctrl_apply_state(PINCTRL_DT_DEV_CONFIG_GET(DT_NODELABEL(isp)),
+			    PINCTRL_STATE_DEFAULT);
+
+	return 0;
+}
+
+SYS_INIT(nxp_rt500_late_init, PRE_KERNEL_2, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
Seems like you would want to check the status or this will cause a build failure on any board using this soc which does not define or enable the ISP pins?

Reply:
This should not have an impact on boards that do not define the pinctrl setting for their board.

Reply:
@thedjnK, I have added a build time check to avoid these extra functions from being called on platforms that do not require reconfiguring the boot pins.